cmd/crank/beta/convert/pipelinecomposition/cmd.go

```diff
@@ -71,34 +85,62 @@ func (c *Cmd) AfterApply() error {
 }
 
 // Run converts a classic Composition to a function pipeline Composition.
-func (c *Cmd) Run() error {
-	data, err := io.Read(c.fs, c.InputFile)
+func (c *Cmd) Run(k *kong.Context) error {
+	data, err := commonio.Read(c.fs, c.InputFile)
 	if err != nil {
 		return err
 	}
 
-	// Set up schemes for our API types
-	sch := runtime.NewScheme()
-	_ = scheme.AddToScheme(sch)
-	_ = v1.AddToScheme(sch)
+	u := &unstructured.Unstructured{}
 
-	decode := serializer.NewCodecFactory(sch).UniversalDeserializer().Decode
+	if err := yaml.Unmarshal(data, u); err != nil {
+		return errors.Wrap(err, "Unmarshalling Error")
+	}
 
-	oc := &v1.Composition{}
-	_, _, err = decode(data, &v1.CompositionGroupVersionKind, oc)
+	out, err := convertPnTToPipeline(u, c.FunctionPatchAndTransformRef)
 	if err != nil {
-		return errors.Wrap(err, "Decoding Error")
+		return errors.Wrap(err, "Error generating new Composition")
 	}
-
-	_, errs := oc.Validate()
```

Comment:
this was validating the Composition before converting it, I dropped it because it requires unmarshalling into a Composition which would mean dropping the `spec.environment` field right away. Moreover, I'd argue it's not this command's task to validate compositions, it should be just mechanically migrating Compositions independently of their validity. We could validate it after the migration, as it should always produce valid and up-to-date compositions... but how would a user fix an issue on an in-memory Composition? we would need to write it and let the user fix it... but how would they check whether it's valid then? I think we can just keep it simple and avoid validation.

Reply:
Agree, lets keep it simple and let specified validation tools handle the validation